### PR TITLE
API endpoints to list OperatingSystemProfiles

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
Signed-off-by: Waleed Malik <ahmedwaleedmalik@gmail.com>

**What does this PR do / Why do we need it**:
The following endpoints were added to the list OperatingSystemProfiles at a global and cluster level, respectively:
- `/api/v2/seeds/{seed_name}/operatingsystemprofiles`
- `/api/v2/projects/{project_id}/clusters/{cluster_id}/operatingsystemprofiles`

**Does this PR close any issues?**:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
Add endpoint to list operating system profiles.
```

### CODE REVIEW

File: pkg/handler/v2/operatingsystemprofile/operatingsystemprofile.go [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+		// Fetch OSPs for all the seeds.
+		var osps []osmv1alpha1.OperatingSystemProfile
+		for _, seed := range seeds {
+			ospList, err := operatingSystemProfile.ListUnsecured(ctx, seed.Namespace)
+			if err != nil {
+				return nil, common.KubernetesErrorToHTTPError(err)
+			}
+
+			osps = append(osps, ospList.Items...)
+		}
```
Comment (lsviben) [inline pkg/handler/v2/operatingsystemprofile/operatingsystemprofile.go]: The seeds will all have the same `kubermatic` namespace. And with this you will only get the OSP from the master cluster, which I guess is not your goal?

I understand that the custom OSPs are on the `kubermatic` namspace of the Seed clusters. So you would need to use the dynamic client that gets set based on the Seed in question. For that you would also need the {seed} in the endpoint.

Or if you want to take the custom OSPs from the master cluster, you dont need to iterate over the seeds, you can just get them from the `kubermatic` namespace. But if the OSPs reside in the master cluster, you will also need a synchronizer between master and seed clusters, which syncs the OSPs from master to all seeds (there is a bunch of similar resources already doing the same).

File: pkg/handler/v2/operatingsystemprofile/operatingsystemprofile.go [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+		userClusterNamespace, err := clusterNamespaceFromContext(ctx, userInfoGetter, req.ProjectID, req.ClusterID)
+		if err != nil {
+			return nil, err
+		}
+
+		ospList, err := operatingSystemProfile.ListUnsecured(ctx, userClusterNamespace)
+		if err != nil {
+			return nil, common.KubernetesErrorToHTTPError(err)
+		}
```
Comment (lsviben) [inline pkg/handler/v2/operatingsystemprofile/operatingsystemprofile.go]: This wont work when the clusters are on Seeds which are not also the Master cluster. You need a dynamic client

File: pkg/handler/v2/operatingsystemprofile/operatingsystemprofile.go (hunk 1 above)
Comment (ahmedwaleedmalik) [reply pkg/handler/v2/operatingsystemprofile/operatingsystemprofile.go]: I'll properly look into this tomorrow. I think I went forward with the PR in a bit of haste. Which was wrong from my side.

Comment (lsviben) [reply pkg/handler/v2/operatingsystemprofile/operatingsystemprofile.go]: Dont worry, we can discuss it tomorrow :)

File: pkg/handler/v2/operatingsystemprofile/operatingsystemprofile_test.go [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+			existingObjects: test.GenDefaultKubermaticObjects(
+				test.GenTestSeed(),
+				test.GenDefaultCluster(),
+			),
+			apiUser:            test.GenDefaultAPIUser(),
+			expectedHTTPStatus: http.StatusOK,
+			expectedResult:     []*apiv2.OperatingSystemProfile{},
+		},
+		{
+			name:      "John cannot list Bob's ApplicationInstallation",
```
Comment (lsviben) [inline pkg/handler/v2/operatingsystemprofile/operatingsystemprofile_test.go]:
```suggestion
			name:      "John cannot list Bob's OSP",
```

**Comment by xrstf** [comment]
/hold

temporarily

**Pull request merged**